src/perl6-debug.nqp

```diff
@@ -476,13 +476,20 @@ sub MAIN(*@ARGS) {
     
     # Set up END block list, which we'll run at exit.
     nqp::bindhllsym('perl6', '@END_PHASERS', []);
-    
+
     # Force loading of the debugger module.
+    my $debugger;
+    if @ARGS[1] ~~ /^\-D/ {
+	$debugger := "-M" ~ nqp::substr(@ARGS[1], 2);
+	@ARGS.shift();
```

Comment:
This seems a little fragile, in so far as:

    perl6-debug -MSomeModule -Ilib -DDebug::Trace foo.p6

Would fail to spot the -DDebug::Trace, because it's not the first option. Maybe scanning through the list until a non-option is found (something not starting with -) would be better. Then splice that argument out (`nqp::splice`).